src/MagnumPlugins/BasisImporter/BasisImporter.cpp

```diff
@@ -657,7 +691,30 @@ template<UnsignedInt dimensions> Containers::Optional<ImageData<dimensions>> Bas
     }
 
     if(isUncompressed) {
-        Trade::ImageData<dimensions> out{pixelFormat(*targetFormat, _state->isSrgb), Math::Vector<dimensions, Int>::pad(Vector3i{size}), Utility::move(dest), ImageFlag<dimensions>(UnsignedShort(_state->imageFlags))};
+        const Magnum::PixelFormat outFormat = pixelFormat(*targetFormat, _state->isSrgb);
+        const Math::Vector<dimensions, Int> outSize = Math::Vector<dimensions, Int>::pad(Vector3i{size});
+        /* All (compressed or uncompressed) formats have a block/pixel size
+           that's a multiple of 4, matching the PixelStorage default alignment
+           of 4. *Except* for RGB16F. Basis transcoding functions only take row
+           stride in pixels, not bytes, so we need to manually copy to a
+           correctly aligned image. */
+        const UnsignedInt rowWidthInBytes = formatSize*size.x();
+        if(rowWidthInBytes % 4 != 0) {
```

Comment:
Basis only accepts row stride in multiples of the pixel size, *not* bytes, which breaks for RGB16F. For some reason I figured an intermediate copy is better than returning an image with `PixelStorage` with alignment of 1. But I just realized some importers (Dds, Ktx) do this, so that's probably the better option?

Another alternative would be to simply not allow RGB16F transcoding. Some GPU APIs don't support this at all, but it's also weird to not have it since Basis always transcodes HDR images to alpha = 1.

Reply:
I'd personally return 1-byte-aligned image, yeah. In that case the users have the option of either aligning themselves if it's actually a problem (which might not be at all, for example when saving the output as another image format) or importing as RGBA16F instead. With the current behavior they'd get an extra alloc + copy that cannot be avoided, which might be a problem with giant images on memory-constrained systems such as the web.

> Another alternative would be to simply not allow RGB16F transcoding

The output isn't always just fed to GPUs, it can be an input to some test, or it can be somewhere in the processing pipeline of an `magnum-imageconverter` invocation, and in both cases the extra flexibility is always useful.